packages/oscal-react-library/src/components/OSCALEditableTextField.tsx

```diff
@@ -66,14 +85,74 @@ function textFieldWithEditableActions(props, reference, inEditState, setInEditSt
   );
 }
 
-export default function OSCALEditableTextField(props) {
+export interface EditableFieldProps {
+  /**
+   * True if in editor mode
+   */
+  isEditable?: boolean;
+  /**
+   * Currently this is the handleFieldSave() function passed down from the OSCALLoader.
+   * All of the funcionality of editing require this function.
+   */
+  onFieldSave?: (...args: any[]) => void;
+  /**
+   * A partial representation of the json object you are patching.
+   */
+  partialRestData?: any;
+}
+
+interface EditableTextFieldProps extends EditableFieldProps {
+  /**
+   * Name of the field currently being edited
+   */
+  fieldName: string;
+  /**
+   * Path of each parent of field.
+   *
+   * @example: ["catalog", "back-matter", "resources"]
+   * to edit a backmatter resource.
+   */
+  editedField: string[] | null;
+  /**
+   * The value of the object being changed via the REST api call.
+   *
+   * @example: backmatter.resources
+   */
+  editedValue?: any;
+  /**
+   * The id (usually a uuid) of the edited value
+   *
+   * @example resource.uuid
+   */
+  editedValueId?: string;
+  /**
+   * The value of the field that is being changed.
+   *
+   * @example resource.title
+   */
+  value?: string;
+  /**
+   * Grid item xs size
+   */
+  size?: number;
+  /**
+   * Size prop of underling TextField
+   */
+  textFieldSize?: string;
+  /**
+   * The variant of the non editable Typography
+   */
+  typographyVariant?: TypographyVariant;
+}
+
+export default function OSCALEditableTextField(props: EditableTextFieldProps) {
   const reference = useRef("reference to text field");
-  const [inEditState, setInEditState] = useState(props.inEditState);
+  const [inEditState, setInEditState] = useState(false);
 
-  return props.canEdit ? (
+  return props.isEditable ? (
     textFieldWithEditableActions(props, reference, inEditState, setInEditState)
   ) : (
-    <Grid item className={props.className}>
+    <Grid item>
       <Typography variant={props.typographyVariant}>{props.value}</Typography>
     </Grid>
```

Comment:
I'm wondering how much value this `Grid` adds. Would removing the `Grid` and just having `props.value` within the `Typography` support the same cause? Or could we use a `Stack` instead?